moxin-frontend/src/my_models/downloaded_files_table.rs

```diff
@@ -544,59 +540,45 @@ pub enum ButtonType {
 #[derive(Live, LiveHook, Widget)]
 pub struct ActionButton {
     #[deref]
-    view: View,
+    deref: CButton,
     #[live]
     type_: ButtonType,
 }
 
 impl Widget for ActionButton {
     fn handle_event(&mut self, cx: &mut Cx, event: &Event, scope: &mut Scope) {
-        let uid = self.widget_uid().clone();
-        match event.hits(cx, self.view.area()) {
-            Hit::FingerDown(_) => {
-                cx.set_key_focus(self.view.area());
-            }
-            Hit::FingerUp(fe) => {
-                if fe.was_tap() {
-                    let action = match self.type_ {
-                        ButtonType::Play => RowAction::PlayClicked,
-                        ButtonType::Resume => RowAction::ResumeClicked,
-                        ButtonType::Info => RowAction::InfoClicked,
-                        ButtonType::Delete => RowAction::DeleteClicked,
-                    };
-                    cx.widget_action(uid, &scope.path, action);
-                }
-            }
-            Hit::FingerHoverIn(_) => {
-                cx.set_cursor(MouseCursor::Hand);
-            }
-            Hit::FingerHoverOut(_) => {
-                cx.set_cursor(MouseCursor::Arrow);
-            }
-            _ => (),
-        }
+        self.deref.handle_event(cx, event, scope);
+        self.widget_match_event(cx, event, scope);
     }
 
     fn draw_walk(&mut self, cx: &mut Cx2d, scope: &mut Scope, walk: Walk) -> DrawStep {
-        self.view.draw_walk(cx, scope, walk)
+        self.deref.draw_walk(cx, scope, walk)
+    }
+}
+
+impl WidgetMatchEvent for ActionButton {
+    fn handle_actions(&mut self, cx: &mut Cx, actions: &Actions, scope: &mut Scope) {
+        if self.deref.tapped(actions) {
+            let uid = self.widget_uid().clone();
+            let action = match self.type_ {
+                ButtonType::Play => RowAction::PlayClicked,
+                ButtonType::Resume => RowAction::ResumeClicked,
+                ButtonType::Info => RowAction::InfoClicked,
+                ButtonType::Delete => RowAction::DeleteClicked,
+            };
+            cx.widget_action(uid, &scope.path, action);
+        }
     }
 }
 
 impl ActionButtonRef {
     pub fn set_visible(&mut self, visible: bool) {
         if let Some(mut inner) = self.borrow_mut() {
-            inner.view.visible = visible;
+            inner.deref.visible = visible;
         }
     }
 }
```

Comment:
Our current convention is to use `inner.view...`instead of `deref`, IMO it's a bit more explicit so that when navigating code I can quickly see over what we're operating and I don't need to go check separately what we're derefing. I think `deref` is more of a technicality than semantics.